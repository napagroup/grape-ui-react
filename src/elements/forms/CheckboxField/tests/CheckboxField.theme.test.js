import React from 'react';
import 'jest-styled-components';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { CheckboxField } from '..';

const colorOptions = [
  { label: 'Red', value: 'red' },
  { label: 'Yellow', value: 'yellow' },
  { label: 'Green', value: 'green' },
];
const selectedValues = ['red', false, false];

describe('CheckboxField using Theme color', () => {
  it('should return a CheckboxField object with custom color', () => {
    const theme = {
      colors: {
        grapeSoda: {
          base: 'hsl(325, 84.6%, 28%)',
          dark: 'hsl(305, 33.9%, 23.7%)',
          light: 'hsl(313, 67.8%, 47.5%)',
        },
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <CheckboxField bg="grapeSoda.dark" color="grapeSoda.light" controlId="checkbox" labelText="Color" name="Color" options={colorOptions} values={selectedValues}>Lorem Ipsum</CheckboxField>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('CheckboxField using Theme Font Family', () => {
  it('should return a CheckboxField object with a custom font family', () => {
    const theme = {
      fonts: {
        trueSpace: '"DejaVu Sans Mono", "Bitstream Vera Sans Mono", Courier',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <CheckboxField controlId="checkbox" fontFamily="trueSpace" labelText="Color" name="Color" options={colorOptions} values={selectedValues}>Lorem Ipsum</CheckboxField>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should return a CheckboxField object with a custom base font family defined', () => {
    const theme = {
      fonts: {
        base: 'Nunito, sans-serif',
      },
    };
    const element = (
      <ThemeProvider theme={theme}>
        <CheckboxField
          color="red"
          controlId="checkbox"
          labelText="Color"
          name="Color"
          options={colorOptions}
          sm
          values={selectedValues}
        >
          This should be Nunito and red
        </CheckboxField>
      </ThemeProvider>
    );
    const component = renderer.create(element);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
