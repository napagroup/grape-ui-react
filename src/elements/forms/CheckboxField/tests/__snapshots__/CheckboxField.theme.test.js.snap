// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxField using Theme Font Family should return a CheckboxField object with a custom base font family defined 1`] = `
.c0 {
  position: relative;
  background-color: transparent;
  color: inherit;
  padding-bottom: 16px;
  padding-top: 4px;
  width: 100%;
}

.c1 {
  background-color: hsl(0,0%,100%);
  color: inherit;
  font-family: Nunito,sans-serif;
  font-size: 0.8rem;
  font-weight: inherit;
  height: auto;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  position: relative;
  top: auto;
  left: 8px;
  padding-left: 8px;
  padding-right: 8px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c6 {
  box-sizing: border-box;
}

.c2 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  display: inline-block;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-right: 1rem;
  cursor: pointer;
}

.c5 {
  margin: 0;
  margin-left: 1rem;
  margin-right: 0.5rem;
  cursor: pointer;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="checkbox"
  >
    Color
  </label>
  <div
    className="c2"
  >
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="red"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[0]"
          onChange={[Function]}
          type="checkbox"
          value="red"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          This should be Nunito and red
        </input>
        <div
          className="c6"
        >
          Red
        </div>
      </div>
    </label>
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="yellow"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[1]"
          onChange={[Function]}
          type="checkbox"
          value="yellow"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          This should be Nunito and red
        </input>
        <div
          className="c6"
        >
          Yellow
        </div>
      </div>
    </label>
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="green"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[2]"
          onChange={[Function]}
          type="checkbox"
          value="green"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          This should be Nunito and red
        </input>
        <div
          className="c6"
        >
          Green
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`CheckboxField using Theme Font Family should return a CheckboxField object with a custom font family 1`] = `
.c0 {
  position: relative;
  background-color: transparent;
  color: inherit;
  padding-bottom: 16px;
  padding-top: 4px;
  width: 100%;
}

.c1 {
  background-color: hsl(0,0%,100%);
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  height: auto;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  position: relative;
  top: auto;
  left: 8px;
  padding-left: 8px;
  padding-right: 8px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c6 {
  box-sizing: border-box;
}

.c2 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  display: inline-block;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-right: 1rem;
  cursor: pointer;
}

.c5 {
  margin: 0;
  margin-left: 1rem;
  margin-right: 0.5rem;
  cursor: pointer;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="checkbox"
  >
    Color
  </label>
  <div
    className="c2"
  >
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="red"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[0]"
          onChange={[Function]}
          type="checkbox"
          value="red"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          Lorem Ipsum
        </input>
        <div
          className="c6"
        >
          Red
        </div>
      </div>
    </label>
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="yellow"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[1]"
          onChange={[Function]}
          type="checkbox"
          value="yellow"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          Lorem Ipsum
        </input>
        <div
          className="c6"
        >
          Yellow
        </div>
      </div>
    </label>
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="green"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[2]"
          onChange={[Function]}
          type="checkbox"
          value="green"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          Lorem Ipsum
        </input>
        <div
          className="c6"
        >
          Green
        </div>
      </div>
    </label>
  </div>
</div>
`;

exports[`CheckboxField using Theme color should return a CheckboxField object with custom color 1`] = `
.c0 {
  position: relative;
  background-color: transparent;
  color: inherit;
  padding-bottom: 16px;
  padding-top: 4px;
  width: 100%;
}

.c1 {
  background-color: hsl(305,33.9%,23.7%);
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: 0.8rem;
  font-weight: inherit;
  height: auto;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 0.8;
  position: relative;
  top: auto;
  left: 8px;
  padding-left: 8px;
  padding-right: 8px;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
}

.c6 {
  box-sizing: border-box;
}

.c2 {
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c4 {
  -webkit-align-items: baseline;
  -webkit-box-align: baseline;
  -ms-flex-align: baseline;
  align-items: baseline;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  background-color: transparent;
  color: inherit;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif,"Apple Color Emoji","Segoe UI Emoji","Segoe UI Symbol";
  font-size: inherit;
  display: inline-block;
  -webkit-letter-spacing: inherit;
  -moz-letter-spacing: inherit;
  -ms-letter-spacing: inherit;
  letter-spacing: inherit;
  line-height: 1.5;
  text-align: inherit;
  -webkit-text-decoration: inherit;
  text-decoration: inherit;
  margin-right: 1rem;
  cursor: pointer;
}

.c5 {
  margin: 0;
  margin-left: 1rem;
  margin-right: 0.5rem;
  cursor: pointer;
}

<div
  className="c0"
>
  <label
    className="c1"
    htmlFor="checkbox"
  >
    Color
  </label>
  <div
    className="c2"
  >
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="red"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[0]"
          onChange={[Function]}
          type="checkbox"
          value="red"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          Lorem Ipsum
        </input>
        <div
          className="c6"
        >
          Red
        </div>
      </div>
    </label>
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="yellow"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[1]"
          onChange={[Function]}
          type="checkbox"
          value="yellow"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          Lorem Ipsum
        </input>
        <div
          className="c6"
        >
          Yellow
        </div>
      </div>
    </label>
    <label
      className="c3"
      id="checkbox"
      options={
        Array [
          Object {
            "label": "Red",
            "value": "red",
          },
          Object {
            "label": "Yellow",
            "value": "yellow",
          },
          Object {
            "label": "Green",
            "value": "green",
          },
        ]
      }
      value="green"
      values={
        Array [
          "red",
          false,
          false,
        ]
      }
    >
      <div
        className="c4"
      >
        <input
          className="c5"
          disabled={false}
          id="checkbox"
          name="Color[2]"
          onChange={[Function]}
          type="checkbox"
          value="green"
          values={
            Array [
              "red",
              false,
              false,
            ]
          }
        >
          Lorem Ipsum
        </input>
        <div
          className="c6"
        >
          Green
        </div>
      </div>
    </label>
  </div>
</div>
`;
